Add 'build-test' and 'runtest-TESTS' targets to Makefile, to build and run tests
cross-compiled.

Signed-off-by: Tudor Florea <tudor.florea@enea.com>
Upstream-Status: Pending
---
Index: Python-2.7.14/Makefile.pre.in
===================================================================
--- Python-2.7.14.orig/Makefile.pre.in
+++ Python-2.7.14/Makefile.pre.in
@@ -846,14 +846,19 @@ $(LIBRARY_OBJS) $(MODOBJS) Modules/pytho
 
 .PHONY: test testall testuniversal buildbottest pythoninfo
 
-TESTOPTS=	-l $(EXTRATESTOPTS)
+TESTOPTS=	-l -v $(EXTRATESTOPTS)
 TESTPROG=	$(srcdir)/Lib/test/regrtest.py
-TESTPYTHON=	$(RUNSHARED) ./$(BUILDPYTHON) -Wd -3 -E -tt $(TESTPYTHONOPTS)
-test:		@DEF_MAKE_RULE@ platform
-		-find $(srcdir)/Lib -name '*.py[co]' -print | xargs rm -f
+TESTPYTHON=	$(RUNSHARED) $(BUILDPYTHON) -Wd -3 -E -tt $(TESTPYTHONOPTS)
+test:		build-test
+		$(MAKE) runtest-TESTS
+
+runtest-TESTS:
+		-find $(srcdir) -name '*.py[co]' -print | xargs rm -f
 		-$(TESTPYTHON) $(TESTPROG) $(TESTOPTS)
 		$(TESTPYTHON) $(TESTPROG) $(TESTOPTS)
 
+build-test:	@DEF_MAKE_RULE@ platform
+
 testall:	@DEF_MAKE_RULE@ platform
 		-find $(srcdir)/Lib -name '*.py[co]' -print | xargs rm -f
 		$(TESTPYTHON) $(srcdir)/Lib/compileall.py
